src/Data/HashMap/Linear.hs

```diff
@@ -59,218 +58,185 @@ import qualified Prelude
 -- # Core Data Types
 --------------------------------------------------
 
--- | Robin values are triples of the key, value and PSL
--- (the probe sequence length).
-type RobinVal k v = (k, v, PSL)
-
--- | A probe sequence length
-newtype PSL = PSL Int
-  deriving (Prelude.Eq, Prelude.Ord, Prelude.Num, Prelude.Show)
-
--- | An array of Robin values
-type RobinArr k v = Array (RobinVal k v)
-
--- | At minimum, we need to store hashable
--- and identifiable keys
-type Keyed k = (Eq k, Hashable k)
-
 -- | A mutable hashmap with a linear API
 data HashMap k v where
   -- | HashMap (size,count) array-of-robin-values
   -- The size is the amount of memory the array takes up.
   -- The count is the number of stored mappings.
   -- Our sparseness invariant: count*3 <= size.
-  HashMap :: (Int, Int) -> RobinArr k v #-> HashMap k v
+  HashMap :: Size -> Count -> RobinArr k v #-> HashMap k v
```

Comment:
A note for later, but you probably want the size and count to be unpacked. It's likely to be better for performance.